### PR TITLE
#12825 - Behavior 'Include contacts/cases from other jurisdictions' c…

### LINKED ISSUE #12825: Behavior 'Include contacts/cases from other jurisdictions' checkbox is not consistent between the web and mobile app/ between cases and contacts
### Bug Description

Behavior 'Include contacts/cases from other jurisdictions' checkbox is not consistent between the web and mobile app/ between cases and contacts.

For a case owned, but of another disease than the user's limited disease, the case is displayed in the web app only when the user checks the checkbox 'Include cases from other jurisdictions'. But in the mobile app, it is displayed both in the main list and when the user checks the  'Include cases from other jurisdictions'.

For contacts, both in the web and mobile app, if the user owns a contact of a different disease, they can not see it in the web or mobile app either in the main list or when checking the 'Include contacts from other jurisdictions' checkbox.

### Reproduction Steps

In the web app:
I: A Default Surveillance Officer limited to Anthrax
In the web app:
1.Log in with the user mentioned above;
2.Create a case, a contact;
3.Log out and log in with an admin;
4.Change the limited disease of the Surveillance Officer to Congenital Rubella;
5.Log back in with the Surveillance Officer and try to find all of the entities of Anthrax that the user is the reporter of.
Actual behavior:
For case - the user can find the case if they check the checkbox 'Include cases from other jurisdictions';
For contact - the contact is not visible in the main list, nor is it visible when the user checks the 'Include contacts from other jurisdictions'.

In the mobile app:
II: A Default Surveillance Officer limited to Anthrax
In the web app:
1.Repeat the same steps as above until step 3;
2. Log in the mobile app with the Surveillance Officer and do a full sync then log out;
3. In the web app, log in with an Admin and change the limited disease of the Surveillance Officer to Congenital Rubella;
4. In the mobile app, trigger a re-synchronize data and then log out and back in (I do this to make sure that the user rights are updated);
5. After that try to find the entities that you are the reporter of that have the disease Anthrax.
Actual behavior:
For cases - the case is visible in the main list and NOT ONLY when the user checks the checkbox for 'Include cases from other jurisdictions';
For contact - the contact is not visible in the main list, nor is it visible when the user checks the 'Include contacts from other jurisdictions'.

### Expected Behavior

The behavior of displaying owned entities of a different disease than the user's limited disease should be consistent - the entities should be displayed irrespective of the "Include cases/contacts from other jurisdictions" checkbox.

### Affected Area(s)

Web app, Android app

### SORMAS Version

1.92.0; 1.93.0

### Android version/Browser

Chrome

### User Role(s)

Default Environment Surveillance User limited to a disease

### Server URL

local machine 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
…heckbox is not consistent between the web and mobile app/ between cases and contacts

Fixes #12825

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12919

File: sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java [hunk 1]
```diff
@@ -883,6 +886,66 @@ public void testGetIndexListByARestrictedAccessToAssignedEntities() {
...
+			surveillanceOfficer.toReference(),
+			cazePerson4.toReference(),
+			Disease.CORONAVIRUS,
+			CaseClassification.PROBABLE,
+			InvestigationStatus.PENDING,
+			new Date(),
+			rdcf1);
+
+		loginWith(nationalAdmin);
+		assertEquals(4, getCaseFacade().getIndexList(null, 0, 100, null).size());
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java:934]: there is no need to assert that for national admin the get index list works, it is already tested in other tests

File: sormas-backend/src/test/java/de/symeda/sormas/backend/contact/ContactFacadeEjbTest.java [hunk 2]
```diff
@@ -1036,6 +1038,62 @@ public void testGetIndexListByARestrictedAccessToAssignedEntities() {
...
+			new Date(),
+			new Date(),
+			Disease.CORONAVIRUS,
+			rdcf2);
+
+		PersonDto contactPerson4 = creator.createPerson("Contact3", "Person3");
+		final ContactDto contact4 = creator
+			.createContact(surveillanceOfficer.toReference(), null, contactPerson4.toReference(), null, new Date(), new Date(), Disease.EVD, rdcf2);
+
+		loginWith(surveillanceOfficer);
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/contact/ContactFacadeEjbTest.java:1078]: the asserts for a normal surveillance officer are good to go into a separate test

this test is called `testGetIndexListWithLimitedDisease` so please test only what it supposed to be tested

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java:934]: code removed

File: sormas-backend/src/test/java/de/symeda/sormas/backend/contact/ContactFacadeEjbTest.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/contact/ContactFacadeEjbTest.java:1078]: removed the asserts that exist in other tests. Only those connected to limited disease are left

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12919

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**